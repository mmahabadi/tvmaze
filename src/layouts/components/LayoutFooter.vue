<script setup lang="ts">
const year = new Date().getFullYear()
</script>
<template>
  <div class="footer">Copyright Â© {{ year }} TVMAZE</div>
</template>
<style scoped>
.footer {
  @apply mt-9 flex h-16 justify-center items-center bg-gray-900;
}
</style>
