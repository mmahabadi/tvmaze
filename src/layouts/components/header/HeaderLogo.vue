<template>
  <a href="/">
    <img src="@/assets/logo.png" alt="logo" class="w-full h-12 object-contain" />
  </a>
</template>
