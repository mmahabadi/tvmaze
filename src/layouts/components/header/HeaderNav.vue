<template>
  <a class="hover:text-subMain transitions text-white" href="/movies">Movies</a
  ><a class="hover:text-subMain transitions text-white" href="/about-us">About Us</a
  ><a class="hover:text-subMain transitions text-white" href="/contact-us">Contact Us</a
  ><a class="hover:text-subMain transitions text-white" href="/login"
    ><svg
      stroke="currentColor"
      fill="none"
      stroke-width="0"
      viewBox="0 0 24 24"
      class="w-8 h-8"
      height="1em"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M16 7C16 9.20914 14.2091 11 12 11C9.79086 11 8 9.20914 8 7C8 4.79086 9.79086 3 12 3C14.2091 3 16 4.79086 16 7ZM14 7C14 8.10457 13.1046 9 12 9C10.8954 9 10 8.10457 10 7C10 5.89543 10.8954 5 12 5C13.1046 5 14 5.89543 14 7Z"
        fill="currentColor"
      ></path>
      <path
        d="M16 15C16 14.4477 15.5523 14 15 14H9C8.44772 14 8 14.4477 8 15V21H6V15C6 13.3431 7.34315 12 9 12H15C16.6569 12 18 13.3431 18 15V21H16V15Z"
        fill="currentColor"
      ></path></svg></a
  ><a
    class="e=>{let{isActive:s}=e;return s?&quot;text-subMain&quot;:&quot;hover:text-subMain transitions text-white&quot;} relative"
    href="/favorites"
    ><svg
      stroke="currentColor"
      fill="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      class="w-6 h-6"
      height="1em"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
      ></path>
    </svg>
    <div
      class="w-5 h-5 flex-colo rounded-full text-xs bg-subMain text-white absolute -top-5 -right-1"
    >
      3
    </div></a
  >
</template>
