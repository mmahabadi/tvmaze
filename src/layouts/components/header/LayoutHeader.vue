<template>
  <header>
    <div class="container">
      <div class="col-span-1 lg:block hidden"><slot name="logo"></slot></div>

      <div class="col-span-3">
        <slot name="search"></slot>
      </div>
      <div class="header-inner">
        <slot></slot>
      </div>
    </div>
  </header>
</template>
<style>
header {
  @apply shadow-md sticky top-0 z-20 backdrop-blur-sm bg-gray-900/30;

  .container {
    @apply mx-auto py-6 px-2 lg:grid gap-10 grid-cols-7 justify-between items-center;

    .header-inner {
      @apply col-span-3 font-medium text-sm hidden xl:gap-14 2xl:gap-20 justify-between lg:flex xl:justify-end items-center;
    }
  }
}
</style>
