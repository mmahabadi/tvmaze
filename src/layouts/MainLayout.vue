<script setup lang="ts">
import LayoutHeader from './components/header/LayoutHeader.vue'
import HeaderLogo from './components/header/HeaderLogo.vue'
import HeaderSearch from './components/header/HeaderSearch.vue'
import HeaderNav from './components/header/HeaderNav.vue'
import LayoutFooter from './components/LayoutFooter.vue'
import Loading from '@/@ui/components/LoadingOverlay.vue'

defineProps<{ loading: boolean }>()
</script>
<template>
  <main class="bg-black text-white">
    <Loading v-if="loading" />
    <div v-else>
      <LayoutHeader>
        <template v-slot:logo>
          <HeaderLogo />
        </template>
        <template v-slot:search>
          <HeaderSearch />
        </template>

        <HeaderNav />
      </LayoutHeader>
      <slot name="slider"></slot>
      <div class="container mx-auto py-6 px-2">
        <slot></slot>
      </div>
      <LayoutFooter />
    </div>
  </main>
</template>
