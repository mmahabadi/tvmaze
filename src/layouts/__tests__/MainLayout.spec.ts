import { mount, type VueWrapper } from '@vue/test-utils'
import { describe, expect, beforeEach, it } from 'vitest'
import MainLayout from '@/layouts/MainLayout.vue'
import LayoutHeader from '@/layouts/components/header/LayoutHeader.vue'
import HeaderLogo from '@/layouts/components/header/HeaderLogo.vue'
import HeaderNav from '@/layouts/components/header/HeaderNav.vue'
import LayoutFooter from '@/layouts/components/LayoutFooter.vue'
import Loading from '@/@ui/components/LoadingOverlay.vue'

describe('MainLayout', () => {
  let wrapper: VueWrapper<MainLayout>

  beforeEach(() => {
    wrapper = mount(MainLayout, {
      props: {
        loading: false
      },
      global: {
        stubs: {
          LayoutHeader,
          HeaderLogo,
          HeaderSearch: true,
          HeaderNav,
          LayoutFooter,
          Loading,
          RouterLink: true
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('loading overlay is not rendered by default', () => {
    expect(wrapper.findComponent(Loading).exists()).toBe(false)
  })

  it('renders loading overlay when loading prop is true', async () => {
    await wrapper.setProps({ loading: true })

    expect(wrapper.findComponent(Loading).exists()).toBe(true)
  })
})
