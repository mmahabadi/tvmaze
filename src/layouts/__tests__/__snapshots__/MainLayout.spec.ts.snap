// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > renders correctly 1`] = `
"<main data-v-22686b16="" class="main">
  <div data-v-22686b16="" class="scr">
    <header data-v-22686b16="">
      <div class="container">
        <div class="col-span-1 lg:block hidden">
          <router-link-stub data-v-22686b16="" to="/" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
        </div>
        <div class="col-span-3">
          <header-search-stub data-v-22686b16=""></header-search-stub>
        </div>
        <div class="header-inner"><a data-v-1dc335c6="" class="nav-item" href="">Shows</a><a data-v-1dc335c6="" class="nav-item" href="">About Us</a><a data-v-1dc335c6="" class="nav-item" href="">Contact Us</a></div>
      </div>
    </header>
    <div data-v-22686b16="" class="container"></div>
    <div data-v-c8ff7694="" data-v-22686b16="" class="footer">Copyright Â© 2024 TVMAZE</div>
  </div>
</main>"
`;
